juju/interface.py

```diff
@@ -0,0 +1,48 @@
+from juju.framework import Event, EventsBase, Object
+from juju.charm import (
+    RelationJoinedEvent,
+    RelationChangedEvent,
+    RelationDepartedEvent,
+    RelationBrokenEvent,
+)
+
+
+class Endpoint:
+    def __init__(self, interface_type):
+        self.interface_type = interface_type
+        self.name = None
+
+    def __set_name__(self, charm, name):
+        self.name = name
+
+    def __get__(self, charm, _):
+        if charm is None:
+            return self
+        else:
+            return self.interface_type(charm, self.name)
+
+
+class InterfaceEvents(EventsBase):
+    joined = Event(RelationJoinedEvent)
+    changed = Event(RelationChangedEvent)
+    departed = Event(RelationDepartedEvent)
+    broken = Event(RelationBrokenEvent)
+
+
+class InterfaceBase(Object):
```

Comment:
Object should have a better name if this is going to be common, to avoid confusion with Python's common use of the object builtin. Even FrameworkObject would be better.